apollo-router/src/plugins/rhai.rs

```diff
@@ -164,386 +270,545 @@ impl Plugin for Rhai {
+                    .map_err(|e: BoxError| e.to_string().into())
+            })
+            .register_indexer_set_result(|x: &mut Context, key: &str, value: Dynamic| {
+                x.insert(key, value)
+                    .map(|v: Option<Dynamic>| v.unwrap_or_else(|| Dynamic::from(())))
+                    .map_err(|e: BoxError| e.to_string())?;
+                Ok(())
+            })
+            // Register get for Header Name/Value from a tuple pair
+            .register_get("name", |x: &mut (Option<HeaderName>, HeaderValue)| {
+                x.0.clone()
+            })
+            .register_get("value", |x: &mut (Option<HeaderName>, HeaderValue)| {
+                x.1.clone()
+            })
+            // Request.query
+            .register_get("query", |x: &mut Request| {
+                x.query.clone().map_or(Dynamic::from(()), Dynamic::from)
+            })
+            .register_set("query", |x: &mut Request, value: &str| {
+                x.query = Some(value.to_string());
+            })
+            // Request.operation_name
+            .register_get("operation_name", |x: &mut Request| {
+                x.operation_name
+                    .clone()
+                    .map_or(Dynamic::from(()), Dynamic::from)
+            })
+            .register_set("operation_name", |x: &mut Request, value: &str| {
+                x.operation_name = Some(value.to_string());
+            })
+            // Request.variables
+            .register_get_result("variables", |x: &mut Request| {
+                to_dynamic(x.variables.clone())
+            })
+            /* XXX CANNOT DO BECAUSE variables is Arc
+            .register_set_result("variables", |x: &mut Request, om: Map| {
+                x.variables = from_dynamic(om)
+            })
+            */
+            // Request.extensions
+            .register_get_result("extensions", |x: &mut Request| {
+                to_dynamic(x.extensions.clone())
+            })
+            .register_set_result("extensions", |x: &mut Request, om: Map| {
+                let v = rhai::format_map_as_json(&om);
+                x.extensions = serde_json::from_str(&v).map_err(|e| e.to_string())?;
+                Ok(())
+            })
+            // Register a series of logging functions
+            .register_fn("log_trace", |x: &str| {
+                tracing::trace!("{}", x);
+            })
+            .register_fn("log_debug", |x: &str| {
+                tracing::debug!("{}", x);
+            })
+            .register_fn("log_info", |x: &str| {
+                tracing::info!("{}", x);
+            })
+            .register_fn("log_warn", |x: &str| {
+                tracing::warn!("{}", x);
+            })
+            .register_fn("log_error", |x: &str| {
+                tracing::error!("{}", x);
+            })
+            // Register a function for printing to stderr
+            .register_fn("eprint", |x: &str| {
```

Comment:
```suggestion
            .register_fn("eprintln", |x: &str| {
```

Reply:
I'm following the `rhai` convention here rather than the `rust` convention. The `Rhai` `print()` function maps to`println!()`, so I figured that `eprint()` would be a good name for `eprintln!()`.